drivers/syscon/efuse_bflb.c

```diff
@@ -0,0 +1,268 @@
+#endif
+			(1 << EF_CTRL_EF_IF_AUTO_RD_EN_POS) |
+			(0 << EF_CTRL_EF_IF_POR_DIG_POS) |
+			(1 << EF_CTRL_EF_IF_0_INT_CLR_POS) |
+			(0 << EF_CTRL_EF_IF_0_RW_POS) |
+			(1 << EF_CTRL_EF_IF_0_TRIG_POS);
+	sys_write32(tmp, config->addr + EF_CTRL_EF_IF_CTRL_0_OFFSET);
+	system_clock_delay_32M_ms(5);
+
+	/* wait for read to complete */
+	do {
+		system_clock_delay_32M_ms(1);
+		tmp = sys_read32(config->addr + EF_CTRL_EF_IF_CTRL_0_OFFSET);
+	} while ((tmp & EF_CTRL_EF_IF_0_BUSY_MSK) ||
+		!(tmp && EF_CTRL_EF_IF_0_AUTOLOAD_DONE_MSK));
+
+	/* do a 'ahb clock' setup */
+	tmp =	EF_CTRL_EFUSE_CTRL_PROTECT |
+			(EF_CTRL_OP_MODE_AUTO << EF_CTRL_EF_IF_0_MANUAL_EN_POS) |
+			(EF_CTRL_PARA_DFT << EF_CTRL_EF_IF_0_CYC_MODIFY_POS) |
+#if defined(CONFIG_SOC_SERIES_BL60X)
+			(EF_CTRL_SAHB_CLK << EF_CTRL_EF_CLK_SAHB_DATA_SEL_POS) |
+#endif
+			(1 << EF_CTRL_EF_IF_AUTO_RD_EN_POS) |
+			(0 << EF_CTRL_EF_IF_POR_DIG_POS) |
+			(1 << EF_CTRL_EF_IF_0_INT_CLR_POS) |
+			(0 << EF_CTRL_EF_IF_0_RW_POS) |
+			(0 << EF_CTRL_EF_IF_0_TRIG_POS);
+
+	sys_write32(tmp, config->addr + EF_CTRL_EF_IF_CTRL_0_OFFSET);
+}
+
+static void efuse_bflb_cache(const struct device *dev)
+{
+	struct efuse_bflb_data *data = dev->data;
+	const struct efuse_bflb_config *config = dev->config;
+	uint32_t tmp = 0;
+	uint8_t old_clock_root;
+
+	tmp = sys_read32(HBN_BASE + HBN_GLB_OFFSET);
+	old_clock_root = (tmp & HBN_ROOT_CLK_SEL_MSK) >> HBN_ROOT_CLK_SEL_POS;
+
+	system_set_root_clock(0);
+	system_clock_settle();
+
+	system_efuse_read(dev);
+	/* reads *must* be 32-bits aligned AND does not work with the method memcpy uses */
+	for (int i = 0; i < config->size / sizeof(uint32_t); i++) {
+		tmp = sys_read32(config->addr + i * 4);
+		data->cache[i * sizeof(uint32_t) + 3] = (tmp & 0xFF000000U) >> 24;
+		data->cache[i * sizeof(uint32_t) + 2] = (tmp & 0x00FF0000U) >> 16;
+		data->cache[i * sizeof(uint32_t) + 1] = (tmp & 0x0000FF00U) >> 8;
+		data->cache[i * sizeof(uint32_t) + 0] = (tmp & 0x000000FFU);
+	}
+
+	system_set_root_clock(old_clock_root);
+	system_clock_settle();
+	data->cached = true;
+}
+
+static int efuse_bflb_read(const struct device *dev, uint16_t reg, uint32_t *val)
+{
+	if (!dev) {
+		return -ENODEV;
+	}
+
+	struct efuse_bflb_data *data = dev->data;
+
+	if (!val) {
+		return -EINVAL;
+	}
+
+	if (!data->cached) {
+		efuse_bflb_cache(dev);
+	}
+
+	*val = *((uint32_t *)&data->cache[reg]);
+	return 0;
+}
+
+static int efuse_bflb_write(const struct device *dev, uint16_t reg, uint32_t val)
+{
+	return -ENOTSUP;
+}
+
+static int efuse_bflb_size(const struct device *dev, size_t *size)
+{
+	const struct efuse_bflb_config *config = dev->config;
+
+	*size = config->size;
+	return 0;
+}
+
+static int efuse_bflb_get_base(const struct device *dev, uintptr_t *addr)
+{
+	if (!dev) {
+		return -ENODEV;
+	}
+
+	struct efuse_bflb_data *data = dev->data;
+
+	*addr = (uint32_t)data->cache;
+	return 0;
+}
+
+static DEVICE_API(syscon, efuse_bflb_api) = {
+	.read = efuse_bflb_read,
+	.write = efuse_bflb_write,
```

Comment:
drop .`write`, APIs already do this check for you.

Reply:
done